File: app/src/main/java/com/dtu/capstone2/ereading/ui/MainActivity.java
package com.dtu.capstone2.ereading.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;

import com.dtu.capstone2.ereading.R;
import com.dtu.capstone2.ereading.ui.account.AccountFragment;
import com.dtu.capstone2.ereading.ui.home.HomeFragment;
import com.dtu.capstone2.ereading.ui.model.MainPage;
import com.dtu.capstone2.ereading.ui.news.NewsFragment;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private ViewPager mViewPagerMain;
    private TabLayout mTabLayout;
    private MainPagerAdapter mMainPagerAdapter;
    private List<MainPage> mListFragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
        initData();

        mMainPagerAdapter = new MainPagerAdapter(getSupportFragmentManager(), mListFragment);
        mViewPagerMain.setAdapter(mMainPagerAdapter);
        mTabLayout.setupWithViewPager(mViewPagerMain);
    }

    private void initView() {
        mViewPagerMain = findViewById(R.id.viewPagerMain);
        mTabLayout = findViewById(R.id.tabLayoutMain);
    }

    private void initData() {
        mListFragment = new ArrayList<>();
        mListFragment.add(new MainPage(new NewsFragment(), "News"));
        mListFragment.add(new MainPage(new HomeFragment(), "Home"));
        mListFragment.add(new MainPage(new AccountFragment(), "Account"));
    }
}

File: app/src/main/java/com/dtu/capstone2/ereading/ui/MainActivity.kt
package com.dtu.capstone2.ereading.ui

import android.os.Bundle
import androidx.viewpager.widget.ViewPager
import com.dtu.capstone2.ereading.R
import com.dtu.capstone2.ereading.ui.base.BaseActivity
import com.dtu.capstone2.ereading.ui.base.TypeTabPager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity(), ViewPager.OnPageChangeListener {
    private lateinit var mMainPagerAdapter: MainPagerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initData()
        initView()
    }

    override fun onPageScrollStateChanged(p0: Int) {
    }

    override fun onPageScrolled(p0: Int, p1: Float, p2: Int) {
    }

    override fun onPageSelected(position: Int) {
        when (position) {
            TypeTabPager.TAB_NEW_FEED.type -> {
                tab_layout_main?.run {
                    getTabAt(0)?.icon?.alpha = 255
                    getTabAt(1)?.icon?.alpha = 100
                }
            }
            TypeTabPager.TAB_ACCOUNT.type -> {
                tab_layout_main?.run {
                    getTabAt(0)?.icon?.alpha = 100
                    getTabAt(1)?.icon?.alpha = 255
                }
            }
        }
    }

    private fun initData() {
        mMainPagerAdapter = MainPagerAdapter(supportFragmentManager)
    }

    private fun initView() {
        view_pager_main?.apply {
            adapter = mMainPagerAdapter
            offscreenPageLimit = 3
            addOnPageChangeListener(this@MainActivity)
        }

        tab_layout_main?.apply {
            setupWithViewPager(view_pager_main)
            getTabAt(TypeTabPager.TAB_NEW_FEED.type)?.setIcon(R.drawable.ic_news_feed_16)
            getTabAt(TypeTabPager.TAB_ACCOUNT.type)?.setIcon(R.drawable.ic_account_16)
            repeat(tabCount) {
                getTabAt(it)?.run {
                    if (it != 0) {
                        icon?.alpha = 100
                    }
                }
            }
        }
    }
}
